contracts/core/Registry.sol

```diff
@@ -1,47 +1,141 @@
+            _grantRole(memberRole, _members[i]);
+        }
+
+        emit IdentityCreated(identityId, name, metadata, identityDetails.attestationAddr);
+
+        identityId++;
+        
+        return identityId;
     }
 
-    // updates the name of the identity
-    // note that this will also change the attestation address, since it's a hash that includes the name
+    /// @notice Updates the name of the identity. 
+    /// @param _identityId The identityId of the identity
+    /// @param _name The new name of the identity
+    /// @dev Only owner can update the name. Also updated the attestation address
     function updateIdentityName(uint _identityId, string memory _name) external {
 
+        bytes32 ownerRole = _generateRole(identityId, RoleType.OWNER);
+
+        if (!hasRole(ownerRole, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+
+        identities[_identityId].name = _name;
+        identities[_identityId].attestationAddr = _generateAttestationAddr(_identityId, _name);
+
+        emit IdentityNameUpdated(_identityId, _name);
     }
 
     // @todo override changing owners (core owner, not contributors) to make sure it updates attestation address
 
-    // update the metadata of the identity
-    // checks ownership role first, probably separate roles for this vs using it (owner vs user?)
-    function updateIdentityMetadata(uint _identityId, string memory _metadata) external {
+    /// @notice update the metadata of the identity
+    /// @param _identityId The identityId of the identity
+    /// @param _metadata The new metadata of the identity
+    /// @dev Only owner or member can update metadata
+    function updateIdentityMetadata(uint _identityId, Metadata memory _metadata) external {
+
+        bytes32 ownerRole = _generateRole(_identityId, RoleType.OWNER);
+        bytes32 memberRole = _generateRole(_identityId, RoleType.MEMBER);
+
+        if (
+            !hasRole(ownerRole, msg.sender) ||
+            !hasRole(memberRole, msg.sender)
+        ) {
+            revert NO_ACCESS_TO_ROLE();
+        }
 
+        identities[_identityId].metadata = _metadata;
+
+        emit IdentityMetadataUpdated(_identityId, _metadata);
     }
 
-    // this will use solmate ROLES to check if the msg.sender is an owner of the project
-    // @todo figure out how to best represent project ownership and details
-    function isOwnerOfProject(uint _identityId, address _owner) external view returns (bool) {
-        
+    /// @notice Returns if the given address is an owner of the identity
+    function isOwnerOfIdentity(uint _identityId, address _owner) external view returns (bool) {
+        bytes32 ownerRole = _generateRole(_identityId, RoleType.OWNER);
+        return hasRole(ownerRole, _owner);
+    }
+
+
+    /// @notice Returns if the given address is an member of the identity
+    function isMemberOfIdentity(uint _identityId, address _owner) external view returns (bool) {
```

Comment:
@zobront Added this as well cause it makes sense to offer isRolecheck for both roles